exporter/splunkhecexporter/integrationtestutils/splunk.go

```diff
@@ -0,0 +1,227 @@
+	jobID := postSearchRequest(user, password, baseURL, searchQuery, startTime, endTime)
+	// wait for search status done == true
+	for i := 0; i < 20; i++ { // limit loop - not allowing infinite looping
+		logger.Println("Checking Search Status ...")
+		isDone := checkSearchJobStatusCode(user, password, baseURL, jobID)
+		if isDone == true {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	// get events
+	results := getSplunkSearchResults(user, password, baseURL, jobID)
+	return results
+}
+
+func getSplunkSearchResults(user string, password string, baseURL string, jobID string) []interface{} {
+	logger := log.New(os.Stdout, "", log.LstdFlags)
+	eventURL := fmt.Sprintf("%s/services/search/jobs/%s/events?output_mode=json", baseURL, jobID)
+	logger.Println("URL: " + eventURL)
+	reqEvents, err := http.NewRequest("GET", eventURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	reqEvents.SetBasicAuth(user, password)
+	respEvents, err := client.Do(reqEvents)
+	if err != nil {
+		panic(err)
+	}
+	defer respEvents.Body.Close()
+	logger.Println("Send Request: Get query status code: " + strconv.Itoa(respEvents.StatusCode))
+
+	bodyEvents, err := io.ReadAll(respEvents.Body)
+	if err != nil {
+		panic(err)
+	}
+
+	var jsonResponseEvents map[string]interface{}
+	err = json.Unmarshal(bodyEvents, &jsonResponseEvents)
+	if err != nil {
+		panic(err)
+	}
+
+	// logger.Println("json Response Events --->")   # debug
+	// logger.Println(jsonResponseEvents)			# debug
+	results := jsonResponseEvents["results"].([]interface{})
+	// logger.Println(results)
+	return results
+}
+
+func checkSearchJobStatusCode(user string, password string, baseURL string, jobID string) interface{} {
+	logger := log.New(os.Stdout, "", log.LstdFlags)
+	checkEventURL := baseURL + "/services/search/jobs/" + jobID + "?output_mode=json"
+	logger.Println("URL: " + checkEventURL)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	checkReqEvents, err := http.NewRequest("GET", checkEventURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	checkReqEvents.SetBasicAuth(user, password)
+	checkResp, err := client.Do(checkReqEvents)
+	if err != nil {
+		panic(err)
+	}
+	defer checkResp.Body.Close()
+	logger.Println("Send Request: Check query status code: " + strconv.Itoa(checkResp.StatusCode))
+	checkBody, err := io.ReadAll(checkResp.Body)
+	if err != nil {
+		panic(err)
+	}
+	var checkJSONResponse map[string]interface{}
+	err = json.Unmarshal(checkBody, &checkJSONResponse)
+	if err != nil {
+		panic(err)
+	}
+	// logger.Println(checkJSONResponse) // debug
+	// Print isDone field from response
+	isDone := checkJSONResponse["entry"].([]interface{})[0].(map[string]interface{})["content"].(map[string]interface{})["isDone"]
+	logger.Printf("Is Splunk Search compleated [isDone flag]: %v\n", isDone)
```

Comment:
I would strongly advise against doing this, if something changes within the response then this will cause a panic.

Reply:
do you mean to remove logger line? or maybe I should change the way how I am getting "isDone" flag?
I need to extract isDone to progress with test execution, I can use regex here but I think current solution is better

Reply:
@MovieStoreGuy means the above line where you cast and extract the value. It's ok to panic, this is a test, but it will hurt. I think that's ok.